### PR TITLE
10907 odata performance improvements

### PR DESCRIPTION
**Pull request by cooperka**
18 hours straight to migrate GW on 16 cores seems excessive... This is an attempt at low-hanging fruit. Total speedup is ~20-40% depending on form structure.

### CODE REVIEW

**Comment by cooperka** [comment]
Failure was a flapper

File: Gemfile [hunk 1]
```diff
@@ -113,13 +113,21 @@ gem "ruby-jmeter", "~> 2.13.4"
...
   gem "thin", "~> 1.7.0"
+
+  # Great for debugging i18n paths. Uncomment temporarily when needed.
+  # gem "i18n-debug", "~> 1.1"
+
+  # Profiling. Uncomment temporarily when needed.
+  # gem "rack-mini-profiler", "~> 2.0"
+  # gem "memory_profiler", "~> 0.9.14" # Append to URL: ?pp=profile-memory
+  # gem "flamegraph", "~> 0.9.5" # Append to URL: ?pp=flamegraph
+  # gem "stackprof", "~> 0.2.15"
```
Comment (smoyth) [inline Gemfile:130]: Cool!

File: app/models/application_record.rb [hunk 2]
```diff
@@ -14,4 +14,12 @@ def self.clone_options(**options)
     end
     class_variable_set("@@clone_options", options)
   end
+
+  # Identical to calling update! but with validate: false.
+  def update_without_validate!(**args)
+    with_transaction_returning_status do
+      assign_attributes(**args)
+      save!(validate: false)
```
Comment (smoyth) [inline app/models/application_record.rb:22]: I don't think ! makes sense if validations are off. You can just do `save`

File: db/migrate/20200508155813_populate_response_cached_json.rb [hunk 3]
```diff
@@ -40,8 +40,8 @@ def cache_responses(responses)
     total = responses.count
     start = Time.zone.now
     num_procs = ENV["NUM_PROCS"] ? ENV["NUM_PROCS"].to_i : Etc.nprocessors
-    Parallel.each_with_index(responses.find_each, in_processes: num_procs,
-                                                  isolation: true) do |response, index|
+    Parallel.each_with_index(responses.each, in_processes: num_procs,
```
Comment (smoyth) [inline db/migrate/20200508155813_populate_response_cached_json.rb:43]: The change to `each` will mean all responses and answers are loaded into memory at once won't it?

Check this out: https://stackoverflow.com/a/57514911

Comment (cooperka) [reply db/migrate/20200508155813_populate_response_cached_json.rb:43]: Theoretically yes, but in practice it changed nothing. I tried on my local machine with the full Health dataset and it ran smoothly either way. I wonder if Parallel batches internally.

Note: this change was necessary to support `order` above.

**Pull request merged**